pkg/config/virtio.go

```diff
@@ -181,6 +220,165 @@ func (dev *VirtioSerial) FromOptions(options []option) error {
 	return dev.validate()
 }
 
+// VirtioInputNew creates a new input device for the virtual machine.
+// The inputType parameter is the type of virtio-input device that will be added
+// to the machine.
+func VirtioInputNew(inputType string) (VirtioDevice, error) {
+	if inputType != VirtioInputPointingDevice && inputType != VirtioInputKeyboardDevice {
+		return nil, fmt.Errorf("Unknown option for virtio-input devices: %s", inputType)
+	}
+
+	return &VirtioInput{
+		InputType: inputType,
+	}, nil
```

Comment:
This could be written as 
```
    dev := &VirtioInput{
		InputType: inputType,
	}
    if err := dev.validate(); err != nil {
        return nil, err
    }
    return dev, nil
```
to avoid repeating that same validation code multiple times.